### PR TITLE
[younow] Add new extractor

### PR DESCRIPTION
**Pull request by Tatsh**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Support for live streams on [YouNow](https://www.younow.com/).

This should close issues #11310 #9432 #9255

### CODE REVIEW

File: youtube_dl/extractor/younow.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+        return {
+            'id': data['user']['profileUrlString'],
+            'is_live': True,
+            'title': data.get('title'),
+            'description': self._og_search_description(webpage),
+            'thumbnail': data.get('awsUrl'),
+            'tags': data.get('tags'),
+            'categories': data.get('tags'),
+            'uploader': data['user']['profileUrlString'],
```
Comment (dstftw) [inline youtube_dl/extractor/younow.py]: None of the optional fields should break the extraction if missing. Read new extractor tutorial.

File: youtube_dl/extractor/younow.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            'uploader_id': data.get('userId'),
+            'uploader_url': 'https://www.younow.com/%s' % (data['user']['profileUrlString'],),
+            'creator': data['user']['profileUrlString'],
+            'view_count': int(data.get('viewers')),
+            'like_count': int(data.get('likes')),
```
Comment (dstftw) [inline youtube_dl/extractor/younow.py]: `utils.int_or_none`.

File: youtube_dl/extractor/younow.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            'formats': [{
+                'url': stream_url,
+                'no_resume': True,
```
Comment (dstftw) [inline youtube_dl/extractor/younow.py]: Has no effect on hls.

File: youtube_dl/extractor/younow.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            )
+        else:
+            raise UnsupportedError('Unsupported stream')
+
+        webpage = self._download_webpage(url, username)
+
+        return {
+            'id': data['user']['profileUrlString'],
+            'is_live': True,
+            'title': data.get('title'),
```
Comment (dstftw) [inline youtube_dl/extractor/younow.py]: Title is mandatory.

File: youtube_dl/extractor/younow.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                'url': data['ytBroadcastId'],
+                'ie_key': YoutubeIE.ie_key(),
+            }
+
+        # Assume HLS is available
+        media = data.get('media')
+        if media:
+            stream_url = 'https://cdn.younow.com/php/api/broadcast/videoPath/hls=1/broadcastId=%s/channelId=%s' % (
+                data.get('broadcastId'),
+                data.get('userId'),
```
Comment (dstftw) [inline youtube_dl/extractor/younow.py]: This fields are mandatory. It's pointless to allow `None`s for them.

**Comment by jaw-sh** [comment]
Bumping. Would like to see.

**Comment by Tatsh** [comment]
@dstftw This is ready to go on my end.

**Pull request closed without merging**